src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -13,23 +13,32 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
 
 /** Servlet responsible for creating new saved matches AND listing matches. */
 @WebServlet("/matches")
 public class MatchServlet extends HttpServlet {
 
-  private static MatchRepository testRepository;
-  private static User testUser;
-
   @Override
-  public void init() {
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+
     NonPersistentMatchRepository repository = new NonPersistentMatchRepository();
-    testUser = repository.addTestData();
-    testRepository = repository;
+    User testUser = repository.addTestData();
+    MatchRepository testRepository = repository;
+
+    //Set MatchRepository and Current User as ServletContext so that they can be accessed by all servlets
+    getServletContext().setAttribute("matchRepository", testRepository);
+    getServletContext().setAttribute("currentUser", testUser);
```

Comment:
One option would be to use the UserService API to determine what user is making the request rather than trying to track the currentUser using a global variable.

Reply:
Yep, I should be changing this in the next modification when I connect the repository to the logged-in user.